packages/flutter/lib/src/material/drawer.dart

```diff
@@ -218,34 +235,72 @@ class DrawerControllerState extends State<DrawerController> with SingleTickerPro
 
   void _move(DragUpdateDetails details) {
     final double delta = details.primaryDelta / _width;
-    switch (Directionality.of(context)) {
-      case TextDirection.rtl:
-        _controller.value -= delta;
-        break;
-      case TextDirection.ltr:
-        _controller.value += delta;
-        break;
+    if (widget.type == DrawerType.START) {
+      if (Scaffold.of(context).isEndSideDrawerOpen() == true) {
```

Comment:
There should be no need to interact with the `Scaffold` here.  The `DrawerController` should not have any knowledge of the `Scaffold` and should work independently of it.

Reply:
Fixed